internal/cmd/beta/network-interface/update/update.go

```diff
@@ -0,0 +1,242 @@
+				`$ stackit beta network-interface update xxx --network-id yyy --allowed-addresses "1.1.1.1,8.8.8.8,9.9.9.9" --labels key=value"`,
+			),
+			examples.NewExample(
+				`Updates a network interface with nic id "xxx" and network-id "yyy" with new name "nic-name-new"`,
+				`$ stackit beta network-interface update xxx --network-id yyy --name nic-name-new"`,
+			),
+			examples.NewExample(
+				`Updates a network interface with nic id "xxx" and network-id "yyy" with security groups removed and with nic-security disabled`,
+				`$ stackit beta network-interface update xxx --network-id yyy --nic-security=false --security-groups """`,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			model, err := parseInput(p, cmd, args)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			if !model.AssumeYes {
+				prompt := fmt.Sprintf("Are you sure you want to update the network interface %q?", model.NicId)
+				err = p.PromptForConfirmation(prompt)
+				if err != nil {
+					return err
+				}
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			resp, err := req.Execute()
+			if err != nil {
+				return fmt.Errorf("create network area: %w", err)
+			}
+
+			return outputResult(p, model, resp)
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Var(flags.UUIDFlag(), networkIdFlag, "Network ID")
+	cmd.Flags().StringSlice(allowedAddressesFlag, nil, "List of allowed IPs")
+	cmd.Flags().StringToString(labelFlag, nil, "Labels are key-value string pairs which can be attached to a network-interface. A label can be provided with the format key=value and the flag can be used multiple times to provide a list of labels")
+	cmd.Flags().StringP(nameFlag, "n", "", "Network interface name")
+	cmd.Flags().BoolP(nicSecurityFlag, "b", defaultEnabled, "If this is set to false, then no security groups will apply to this network interface.")
+	cmd.Flags().StringSlice(securityGroupsFlag, nil, "List of security groups")
+
+	err := flags.MarkFlagsRequired(cmd, networkIdFlag)
+	cobra.CheckErr(err)
+}
+
+func parseInput(p *print.Printer, cmd *cobra.Command, inputArgs []string) (*inputModel, error) {
+	nicId := inputArgs[0]
+	globalFlags := globalflags.Parse(p, cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &errors.ProjectIdError{}
+	}
+
+	allowedAddresses := flags.FlagToStringSlicePointer(p, cmd, allowedAddressesFlag)
+	var allowedAddressesInner []iaas.AllowedAddressesInner
+	if allowedAddresses != nil && len(*allowedAddresses) > 0 {
+		allowedAddressesInner = make([]iaas.AllowedAddressesInner, len(*allowedAddresses))
+		for k, v := range *allowedAddresses {
+			// linter issue
+			value := v
+			allowedAddressesInner[k].String = &value
```

Comment:
Can't this just be `allowedAddressesInner[k].String = &v`?

Reply:
Although maybe `address` is a better name than `v` for this local variable

Reply:
I got a linter issue from CI: 

```
G601: Implicit memory aliasing in for loop. (gosec)
			allowedAddressesInner[k].String = &v
```
This linter issue should disappear with Go 1.22 but I just saw that we are using Go 1.22. I will try to find out why the linter is still complaining about this (the reassigning is a suggested workaround to prevent the linter issue).

Reply:
It seems to be that we need to update the linter at least to version 1.60.3 to prevent this to pop up. When updating locally there are more linter issues raised which needs to be fixed as well (also some of the configured linters are deprecated and need to be replaced/removed).

How is this done on your side? I would do a different PR for this topic in general. We can keep this PR open until the linter is updated in order to not push the workaround within this PR (or not to forget to remove the workaround afterwards).

Reply:
As discussed, I think your suggestion is the best approach. Lets updated the linter, merge that PR and then come back to this issue

Reply:
Linter was updated, I change the corresponding lines accordingly.